pkg/controller/vitessshardreplication/vitessshardreplication_controller.go

```diff
@@ -226,3 +216,29 @@ func (r *ReconcileVitessShard) Reconcile(request reconcile.Request) (reconcile.R
 	reconcileCount.WithLabelValues(metricLabels(vts, err)...).Inc()
 	return result, err
 }
+
+func (r *ReconcileVitessShard) initReplication(ctx context.Context, vts *planetscalev2.VitessShard, wr *wrangler.Wrangler) (reconcile.Result, error){
+	resultBuilder := &results.Builder{}
+
+	// If we have configured the operator to not initialize replication, bail.
+	if !vts.Spec.VitessShardTemplate.Replication.InitializeMaster {
+		return resultBuilder.Result()
+	}
+
+	// Check if we need to initialize the shard.
+	// If it's already initialized, this will be a no-op.
+	// If we are using external MySQL we will bail out early.
+	ismResult, err := r.initShardMaster(ctx, vts, wr)
+	resultBuilder.Merge(ismResult, err)
+
+	// Check if we need to externally reparent
+	// in the case of external MySQL.
+	// If we are not using external MySQL we will bail out early.
+	terResult, err := r.tabletExternallyReparent(ctx, vts, wr)
+	resultBuilder.Merge(terResult, err)
+
+	// Check if we need to start replication on a shard that's been restored
+	// from backup. If it's already initialized, this will be a no-op.
+	irsResult, err := r.initRestoredShard(ctx, vts, wr)
+	return resultBuilder.Merge(irsResult, err)
```

Comment:
Just call Merge without returning, then have a separate return at the end. This maintains symmetry of all the parts of the function, which makes it easy to add/remove/rearrange in the future.

Reply:
Sounds good, done.